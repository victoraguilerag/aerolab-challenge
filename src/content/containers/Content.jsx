import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Immutable from 'immutable'
import { connect } from 'react-redux'
import {
  updateFilter,
  previousPage,
  nextPage
} from '../../actions/index.js'
import ContentOptions from '../components/ContentOptions.jsx'
import ContentFooter from '../components/ContentFooter.jsx'
import ProductsList from '../components/ProductsList.jsx'
import orderProducts from '../../utils/orderProducts.js'

export class Content extends Component {
  handleFilterChange = (e) => {
    this.props.updateFilter(e.target.innerText)
  }

  handleNextPage = (e) => {
    this.props.nextPage()
  }

  handlePreviousPage = (e) => {
    if (this.props.page - 1 > 0) {
      this.props.previousPage()
    }
  }

  render () {
    const {
      coins,
      handleModal,
      products,
      limit,
      filters,
      filter,
      page
    } = this.props
    const prev = (page > 1)
    const next = (products.size > limit * page)
    const pageLength = page * limit
    const pageProducts = orderProducts(products, filter).slice((page - 1) * limit, page * limit)

    return (
      <div className="content">
        <ContentOptions
          pageLength={pageLength}
          productsLength={products.size}
          activeFilter={filter}
          filters={filters}
          handleFilterChange={this.handleFilterChange}
          prev={prev}
          handlePreviousPage={this.handlePreviousPage}
          next={next}
          handleNextPage={this.handleNextPage}
        />
        <ProductsList
          products={pageProducts}
          limit={limit}
          page={page}
          coins={coins}
          handleModal={handleModal}
        />
        <ContentFooter
          pageLength={pageLength}
          productsLength={products.size}
          prev={prev}
          handlePreviousPage={this.handlePreviousPage}
          next={next}
          handleNextPage={this.handleNextPage}
        />
      </div>
    )
  }
}

Content.propTypes = {
  updateFilter: PropTypes.func.isRequired,
  previousPage: PropTypes.func.isRequired,
  nextPage: PropTypes.func.isRequired,
  coins: PropTypes.number.isRequired,
  handleModal: PropTypes.func.isRequired,
  products: PropTypes.instanceOf(Immutable.List).isRequired,
  limit: PropTypes.number.isRequired,
  filters: PropTypes.instanceOf(Immutable.List).isRequired,
  filter: PropTypes.string.isRequired,
  page: PropTypes.number.isRequired
}

const mapStateToProps = (state, props) => {
  return {
    products: state.get('products').get('products'),
    limit: state.get('products').get('limit'),
    filters: state.get('products').get('filters'),
    filter: state.get('products').get('filter'),
    page: state.get('products').get('page')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    updateFilter: filter => dispatch(updateFilter(filter)),
    previousPage: () => dispatch(previousPage()),
    nextPage: () => dispatch(nextPage())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Content)
