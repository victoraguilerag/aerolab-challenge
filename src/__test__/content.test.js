import React from 'react'
import { Provider } from 'react-redux'
import Content from '../content/containers/Content.jsx'
import productsStub from '../stub/products.js'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'

const mockStore = configureStore()
const functionMock = () => console.log('mock')
const coins = 123
const handleModal = functionMock

const initialState = {
  selectedItem: '123123',
  limit: 16,
  filters: [],
  filter: 'filter',
  products: productsStub,
  page: 1
}

const store = mockStore(initialState)

test('Content renders', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Content coins={coins} handleModal={handleModal}/>
    </Provider>,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
