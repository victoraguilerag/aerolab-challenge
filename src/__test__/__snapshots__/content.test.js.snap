// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders 1`] = `
<div
  className="content"
>
  <div
    className="content-options"
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottom": "solid 1px #d9d9d9",
        "display": "flex",
        "flexFlow": "column wrap",
        "height": "70px",
        "justifyContent": "flex-start",
        "margin": "0 auto",
        "maxWidth": "1176px",
      }
    }
  >
    <div
      className="options"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="products-count-options"
        style={
          Object {
            "alignItems": "center",
            "color": "#616161",
            "display": "flex",
            "fontSize": "24px",
            "justifyContent": "center",
            "letterSpacing": "-0.15px",
            "lineHeight": "48px",
            "textAlign": "left",
          }
        }
      >
        16 of 4 items
      </div>
      <div
        className="divider"
        style={
          Object {
            "background": "#d9d9d9",
            "height": "49px",
            "margin": "5px",
            "marginLeft": "23px",
            "width": "1px",
          }
        }
      />
      <div
        className="filters"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <p
          className="filters-label"
          style={
            Object {
              "alignItems": "center",
              "color": "#a3a3a3",
              "display": "flex",
              "fontSize": "24px",
              "justifyContent": "center",
              "letterSpacing": "-0.15px",
              "margin": "0",
              "marginLeft": "24px",
              "textAlign": "left",
              "userSelect": "none",
            }
          }
        >
          Sort by:
        </p>
        <ul
          className="filters"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexFlow": "row",
              "listStyle": "none",
              "margin": "0",
              "padding": "0",
            }
          }
        />
      </div>
    </div>
    <div
      className="controls"
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "flex-end",
          "display": "flex",
          "height": "100%",
        }
      }
    />
  </div>
  <div
    className="product-list"
    style={
      Object {
        "display": "flex",
        "flexFlow": "row wrap",
        "justifyContent": "space-around",
        "margin": "0 auto",
        "maxWidth": "1176px",
      }
    }
  >
    <span
      style={
        Object {
          "display": "flex",
          "flexFlow": "row wrap",
          "justifyContent": "space-around",
          "margin": "0 auto",
          "maxWidth": "1176px",
        }
      }
    >
      <div
        className="product-card"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
            "boxShadow": "2px 2px 4px 0 rgba(0,0,0,0.10)",
            "cursor": "pointer",
            "height": "276px",
            "marginTop": "41px",
            "position": "relative",
            "userSelect": "none",
            "width": "276px",
          }
        }
      >
        <div
          className="loader"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": "182px",
              "justifyContent": "center",
              "marginTop": "12px",
              "width": "252px",
            }
          }
        >
          <div>
            <svg
              height="50"
              viewBox="0 0 38 38"
              width="100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <linearGradient
                  id="a"
                  x1="8.042%"
                  x2="65.682%"
                  y1="0%"
                  y2="23.865%"
                >
                  <stop
                    offset="0%"
                    stopColor="#00bfff"
                    stopOpacity="0"
                  />
                  <stop
                    offset="63.146%"
                    stopColor="#00bfff"
                    stopOpacity=".631"
                  />
                  <stop
                    offset="100%"
                    stopColor="#00bfff"
                  />
                </linearGradient>
              </defs>
              <g
                fill="none"
                fillRule="evenodd"
              >
                <g
                  transform="translate(1 1)"
                >
                  <path
                    d="M36 18c0-9.94-8.06-18-18-18"
                    id="Oval-2"
                    stroke="url(#a)"
                    strokeWidth="2"
                  >
                    <animateTransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </path>
                  <circle
                    cx="36"
                    cy="18"
                    fill="#fff"
                    r="1"
                  >
                    <animateTransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </circle>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <img
          alt="Secador de Cabello"
          onError={[Function]}
          onLoad={[Function]}
          src="asdas"
          srcSet="asdas 2x"
          style={
            Object {
              "height": "182px",
              "marginTop": "12px",
              "pointerEvents": "none",
              "width": "252px",
            }
          }
        />
        <div
          className="productUnavailable"
          style={
            Object {
              "alignItems": "center",
              "background": "#616161",
              "borderRadius": "100px",
              "display": "flex",
              "height": "42px",
              "justifyContent": "center",
              "opacity": "0.8",
              "pointerEvents": "none",
              "position": "absolute",
              "right": "12px",
              "top": "12px",
              "width": "142px",
            }
          }
        >
          <p
            className="productUnavailableText"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "14px",
                "letterSpacing": "-0.03px",
                "textAlign": "right",
              }
            }
          >
            You need 2000
          </p>
          <img
            alt="coin"
            className="productUnavailableCoin"
            height="24px"
            src={Object {}}
            style={
              Object {
                "height": "20px",
                "margin": "0 2.5px",
                "width": "20px",
              }
            }
            width="24px"
          />
        </div>
        <div
          className="product-info"
          style={
            Object {
              "bottom": "15px",
              "left": "24px",
              "pointerEvents": "none",
              "position": "absolute",
            }
          }
        >
          <div
            className="product-category"
            style={
              Object {
                "color": "#a3a3a3",
                "fontSize": "16px",
                "letterSpacing": "-0.04px",
                "pointerEvents": "none",
                "textAlign": "left",
              }
            }
          >
            electrodomestico
          </div>
          <div
            className="product-name"
            style={
              Object {
                "color": "#616161",
                "fontSize": "18px",
                "letterSpacing": "-0.04px",
                "pointerEvents": "none",
                "textAlign": "left",
              }
            }
          >
            Secador de Cabello
          </div>
        </div>
      </div>
    </span>
    <span
      style={
        Object {
          "display": "flex",
          "flexFlow": "row wrap",
          "justifyContent": "space-around",
          "margin": "0 auto",
          "maxWidth": "1176px",
        }
      }
    >
      <div
        className="product-card"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
            "boxShadow": "2px 2px 4px 0 rgba(0,0,0,0.10)",
            "cursor": "pointer",
            "height": "276px",
            "marginTop": "41px",
            "position": "relative",
            "userSelect": "none",
            "width": "276px",
          }
        }
      >
        <div
          className="loader"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": "182px",
              "justifyContent": "center",
              "marginTop": "12px",
              "width": "252px",
            }
          }
        >
          <div>
            <svg
              height="50"
              viewBox="0 0 38 38"
              width="100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <linearGradient
                  id="a"
                  x1="8.042%"
                  x2="65.682%"
                  y1="0%"
                  y2="23.865%"
                >
                  <stop
                    offset="0%"
                    stopColor="#00bfff"
                    stopOpacity="0"
                  />
                  <stop
                    offset="63.146%"
                    stopColor="#00bfff"
                    stopOpacity=".631"
                  />
                  <stop
                    offset="100%"
                    stopColor="#00bfff"
                  />
                </linearGradient>
              </defs>
              <g
                fill="none"
                fillRule="evenodd"
              >
                <g
                  transform="translate(1 1)"
                >
                  <path
                    d="M36 18c0-9.94-8.06-18-18-18"
                    id="Oval-2"
                    stroke="url(#a)"
                    strokeWidth="2"
                  >
                    <animateTransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </path>
                  <circle
                    cx="36"
                    cy="18"
                    fill="#fff"
                    r="1"
                  >
                    <animateTransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </circle>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <img
          alt="Algodon"
          onError={[Function]}
          onLoad={[Function]}
          src="asdas"
          srcSet="asdas 2x"
          style={
            Object {
              "height": "182px",
              "marginTop": "12px",
              "pointerEvents": "none",
              "width": "252px",
            }
          }
        />
        <div
          className="productUnavailable"
          style={
            Object {
              "alignItems": "center",
              "background": "#616161",
              "borderRadius": "100px",
              "display": "flex",
              "height": "42px",
              "justifyContent": "center",
              "opacity": "0.8",
              "pointerEvents": "none",
              "position": "absolute",
              "right": "12px",
              "top": "12px",
              "width": "142px",
            }
          }
        >
          <p
            className="productUnavailableText"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "14px",
                "letterSpacing": "-0.03px",
                "textAlign": "right",
              }
            }
          >
            You need 3000
          </p>
          <img
            alt="coin"
            className="productUnavailableCoin"
            height="24px"
            src={Object {}}
            style={
              Object {
                "height": "20px",
                "margin": "0 2.5px",
                "width": "20px",
              }
            }
            width="24px"
          />
        </div>
        <div
          className="product-info"
          style={
            Object {
              "bottom": "15px",
              "left": "24px",
              "pointerEvents": "none",
              "position": "absolute",
            }
          }
        >
          <div
            className="product-category"
            style={
              Object {
                "color": "#a3a3a3",
                "fontSize": "16px",
                "letterSpacing": "-0.04px",
                "pointerEvents": "none",
                "textAlign": "left",
              }
            }
          >
            personal health
          </div>
          <div
            className="product-name"
            style={
              Object {
                "color": "#616161",
                "fontSize": "18px",
                "letterSpacing": "-0.04px",
                "pointerEvents": "none",
                "textAlign": "left",
              }
            }
          >
            Algodon
          </div>
        </div>
      </div>
    </span>
    <span
      style={
        Object {
          "display": "flex",
          "flexFlow": "row wrap",
          "justifyContent": "space-around",
          "margin": "0 auto",
          "maxWidth": "1176px",
        }
      }
    >
      <div
        className="product-card"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
            "boxShadow": "2px 2px 4px 0 rgba(0,0,0,0.10)",
            "cursor": "pointer",
            "height": "276px",
            "marginTop": "41px",
            "position": "relative",
            "userSelect": "none",
            "width": "276px",
          }
        }
      >
        <div
          className="loader"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": "182px",
              "justifyContent": "center",
              "marginTop": "12px",
              "width": "252px",
            }
          }
        >
          <div>
            <svg
              height="50"
              viewBox="0 0 38 38"
              width="100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <linearGradient
                  id="a"
                  x1="8.042%"
                  x2="65.682%"
                  y1="0%"
                  y2="23.865%"
                >
                  <stop
                    offset="0%"
                    stopColor="#00bfff"
                    stopOpacity="0"
                  />
                  <stop
                    offset="63.146%"
                    stopColor="#00bfff"
                    stopOpacity=".631"
                  />
                  <stop
                    offset="100%"
                    stopColor="#00bfff"
                  />
                </linearGradient>
              </defs>
              <g
                fill="none"
                fillRule="evenodd"
              >
                <g
                  transform="translate(1 1)"
                >
                  <path
                    d="M36 18c0-9.94-8.06-18-18-18"
                    id="Oval-2"
                    stroke="url(#a)"
                    strokeWidth="2"
                  >
                    <animateTransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </path>
                  <circle
                    cx="36"
                    cy="18"
                    fill="#fff"
                    r="1"
                  >
                    <animateTransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </circle>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <img
          alt="Time Travel"
          onError={[Function]}
          onLoad={[Function]}
          src="asdas"
          srcSet="asdas 2x"
          style={
            Object {
              "height": "182px",
              "marginTop": "12px",
              "pointerEvents": "none",
              "width": "252px",
            }
          }
        />
        <img
          alt="affortable"
          height="42px"
          src={Object {}}
          style={
            Object {
              "pointerEvents": "none",
              "position": "absolute",
              "right": "12px",
              "top": "12px",
            }
          }
          width="42px"
        />
        <div
          className="product-info"
          style={
            Object {
              "bottom": "15px",
              "left": "24px",
              "pointerEvents": "none",
              "position": "absolute",
            }
          }
        >
          <div
            className="product-category"
            style={
              Object {
                "color": "#a3a3a3",
                "fontSize": "16px",
                "letterSpacing": "-0.04px",
                "pointerEvents": "none",
                "textAlign": "left",
              }
            }
          >
            Transport
          </div>
          <div
            className="product-name"
            style={
              Object {
                "color": "#616161",
                "fontSize": "18px",
                "letterSpacing": "-0.04px",
                "pointerEvents": "none",
                "textAlign": "left",
              }
            }
          >
            Time Travel
          </div>
        </div>
      </div>
    </span>
    <span
      style={
        Object {
          "display": "flex",
          "flexFlow": "row wrap",
          "justifyContent": "space-around",
          "margin": "0 auto",
          "maxWidth": "1176px",
        }
      }
    >
      <div
        className="product-card"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
            "boxShadow": "2px 2px 4px 0 rgba(0,0,0,0.10)",
            "cursor": "pointer",
            "height": "276px",
            "marginTop": "41px",
            "position": "relative",
            "userSelect": "none",
            "width": "276px",
          }
        }
      >
        <div
          className="loader"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": "182px",
              "justifyContent": "center",
              "marginTop": "12px",
              "width": "252px",
            }
          }
        >
          <div>
            <svg
              height="50"
              viewBox="0 0 38 38"
              width="100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <linearGradient
                  id="a"
                  x1="8.042%"
                  x2="65.682%"
                  y1="0%"
                  y2="23.865%"
                >
                  <stop
                    offset="0%"
                    stopColor="#00bfff"
                    stopOpacity="0"
                  />
                  <stop
                    offset="63.146%"
                    stopColor="#00bfff"
                    stopOpacity=".631"
                  />
                  <stop
                    offset="100%"
                    stopColor="#00bfff"
                  />
                </linearGradient>
              </defs>
              <g
                fill="none"
                fillRule="evenodd"
              >
                <g
                  transform="translate(1 1)"
                >
                  <path
                    d="M36 18c0-9.94-8.06-18-18-18"
                    id="Oval-2"
                    stroke="url(#a)"
                    strokeWidth="2"
                  >
                    <animateTransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </path>
                  <circle
                    cx="36"
                    cy="18"
                    fill="#fff"
                    r="1"
                  >
                    <animateTransform
                      attributeName="transform"
                      dur="0.9s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </circle>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <img
          alt="Rick and Morty Season 5 DVD"
          onError={[Function]}
          onLoad={[Function]}
          src="asdas"
          srcSet="asdas 2x"
          style={
            Object {
              "height": "182px",
              "marginTop": "12px",
              "pointerEvents": "none",
              "width": "252px",
            }
          }
        />
        <div
          className="productUnavailable"
          style={
            Object {
              "alignItems": "center",
              "background": "#616161",
              "borderRadius": "100px",
              "display": "flex",
              "height": "42px",
              "justifyContent": "center",
              "opacity": "0.8",
              "pointerEvents": "none",
              "position": "absolute",
              "right": "12px",
              "top": "12px",
              "width": "142px",
            }
          }
        >
          <p
            className="productUnavailableText"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "14px",
                "letterSpacing": "-0.03px",
                "textAlign": "right",
              }
            }
          >
            You need 1000
          </p>
          <img
            alt="coin"
            className="productUnavailableCoin"
            height="24px"
            src={Object {}}
            style={
              Object {
                "height": "20px",
                "margin": "0 2.5px",
                "width": "20px",
              }
            }
            width="24px"
          />
        </div>
        <div
          className="product-info"
          style={
            Object {
              "bottom": "15px",
              "left": "24px",
              "pointerEvents": "none",
              "position": "absolute",
            }
          }
        >
          <div
            className="product-category"
            style={
              Object {
                "color": "#a3a3a3",
                "fontSize": "16px",
                "letterSpacing": "-0.04px",
                "pointerEvents": "none",
                "textAlign": "left",
              }
            }
          >
            entertainment
          </div>
          <div
            className="product-name"
            style={
              Object {
                "color": "#616161",
                "fontSize": "18px",
                "letterSpacing": "-0.04px",
                "pointerEvents": "none",
                "textAlign": "left",
              }
            }
          >
            Rick and Morty Season 5 DVD
          </div>
        </div>
      </div>
    </span>
  </div>
  <div
    className="content-footer"
    style={
      Object {
        "alignItems": "center",
        "borderBottom": "solid 1px #d9d9d9",
        "display": "flex",
        "height": "70px",
        "justifyContent": "flex-start",
        "margin": "0 auto",
        "marginBottom": "100px",
        "marginTop": "59px",
        "maxWidth": "1176px",
        "position": "relative",
      }
    }
  >
    <p
      className="products-count-footer"
      style={
        Object {
          "alignItems": "center",
          "color": "#616161",
          "display": "flex",
          "fontSize": "24px",
          "justifyContent": "center",
          "letterSpacing": "-0.15px",
          "lineHeight": "48px",
          "margin": "0 10px",
          "textAlign": "left",
        }
      }
    >
      16 of 4 items
    </p>
    <div
      className="products-controls"
      style={
        Object {
          "position": "absolute",
          "right": "5px",
        }
      }
    />
  </div>
</div>
`;
